parcels/fieldset.py

```diff
@@ -174,7 +144,8 @@ def add_field(self, field: Field, name: str | None = None):
             raise RuntimeError(f"FieldSet already has a Field with name '{name}'")
         else:
             setattr(self, name, field)
-            self.gridset.add_grid(field)
+            self._gridset_size += 1
+            self._fieldnames.append(name)
```

Comment:
Not 100% sure if it would be as simple as `_gridset_size += 1` since fields could share a grid? at the moment this is tied to `len(self._fieldnames)` so maybe remove `_gridset_size` and update the `gridset_size` property to return `len(self._fieldnames)` then later we could handle duplicate grids if needed?

Reply:
Made the change in https://github.com/OceanParcels/Parcels/pull/1946/commits/81793d980c6adee0c310374be4ae33e750b79886